sdk-cpp/include/do_download_property.h

```diff
@@ -23,69 +22,84 @@ class CDownloadPropertyValueInternal;
     network_token,                  // bool
-    correlation_vector,             // std::string
-    decryption_info,                // std::string
-    integrity_check_info,           // std::string
-    integrity_check_mandatory,      // boolean
+
+    // Available beginning in Windows 20H1 (build 19041)
+    correlation_vector,             // string
+    decryption_info,                // string
+    integrity_check_info,           // string
+    integrity_check_mandatory,      // bool
     total_size_bytes,               // uint64
 
-    // For the COM interface, the following properties are available only in Windows 21H2 (Build Number 22000) and beyond
+    // Available beginning in Windows 21H2 (build 22000)
     disallow_on_cellular,           // bool
-    http_custom_auth_headers,       // std::string
+    http_custom_auth_headers,       // string
+
+    // Available beginning in Windows 22H2 (build 22621)
+    allow_http_to_https_redirect,   // bool
+    non_volatile,                   // bool
 };
 
-class download_property_value
+// Values for download_property::cost_policy
+enum class download_cost_policy : uint32_t
 {
+    always = 0,             // download regardless of cost (foreground default)
+    unrestricted_network,   // pause download on any metered network
+    standard,               // pause download if over or near data limit (background default)
+    no_roaming,             // pause download if roaming
+    no_surcharge,           // pause download if over data limit
+};
 
-/*
-CDownloadImpl is declared as a friend class because it needs to access the platform-specific native value for download_property_value
-The type of the native value is defined in CDownloadPropertyValueInternal, because DO header files are platform agnostic
-This is so any user of the SDK does not have to worry about supplying platform specific compile definitions to use the SDK
-*/
+// Values for download_property::security_flags
+enum class download_security_flags : uint32_t
+{
+    enable_ssl_revocation       = 0x0001, // WINHTTP_ENABLE_SSL_REVOCATION
+    ignore_unknown_ca           = 0x0100, // SECURITY_FLAG_IGNORE_UNKNOWN_CA
+    ignore_cert_wrong_usage     = 0x0200, // SECURITY_FLAG_IGNORE_CERT_WRONG_USAGE
+    ignore_cert_cn_invalid      = 0x1000, // SECURITY_FLAG_IGNORE_CERT_CN_INVALID
+    ignore_cert_date_invalid    = 0x2000, // SECURITY_FLAG_IGNORE_CERT_DATE_INVALID
+    ignore_all_cert_errors      = 0x3300, // SECURITY_FLAG_IGNORE_ALL_CERT_ERRORS
+};
+
+class download_property_value
+{
 friend class details::CDownloadImpl;
 
 public:
-    using status_callback_t = std::function<void(download&, download_status&)>;
-
     download_property_value();
     ~download_property_value() = default;
 
     static std::error_code make(const std::string& val, download_property_value& out);
+    static std::error_code make(const char* val, download_property_value& out) { return make(std::string(val), out); }
```

Comment:
>const char* [](http://example.com/codeflow?start=32&length=11)

Without the const char* overload, doesn't the compiler automatically call the std::string& version?

Reply:
Haven't looked at the code generation, but it picks the wrong overload (probably bool), so we need these overloads.